src/dialects/postgres/query-generator.js

```diff
@@ -869,38 +900,170 @@ export class PostgresQueryGenerator extends AbstractQueryGenerator {
+              (parse_ident(main.confrelid::regclass::text))[
+                array_length(parse_ident(main.confrelid::regclass::text),1)
+              ]                          AS "referencedTableName",
+              JSON_AGG(frn_att.attname) AS "referencedTableColumnNames",
+              conindid::regclass         AS "referencedTableConstraintName"
+
+      FROM
+              -----------------------------------------------
+              -- CONSTRAINTS
+              -----------------------------------------------
+              (
+                select *,                    -- oid column is added in v12, but not included in v10
+                      oid AS constraint_oid, -- specify for version <12 and alias to reduce ambiguity conflict
+                      unnest(conkey)  as ckey, -- unnest for join
+                      unnest(confkey) as cfkey -- unnest for join
+                from pg_catalog.pg_constraint pgc
+              ) AS main
+
+              -----------------------------------------------
+              -- CONSTRAINT-TABLE ATTRIBUTES
+              -----------------------------------------------
+              JOIN  pg_catalog.pg_attribute AS tbl_att
+                    ON (tbl_att.attrelid = conrelid  AND tbl_att.attnum = ckey)
+
+              -----------------------------------------------
+              -- CONSTRAINT-REFERENCE-TABLE ATTRIBUTES
+              -----------------------------------------------
+              JOIN  pg_catalog.pg_attribute AS frn_att
+                    ON (frn_att.attrelid = confrelid AND frn_att.attnum = cfkey)
+
+              -----------------------------------------------
+              -- SCHEMA INFO
+              -----------------------------------------------
+              JOIN  (
+                select  cl.oid, ns.nspname
+                from    pg_class cl
+                join    pg_namespace ns on (cl.relnamespace = ns.oid)
+                where   relname = ${this.escape(tableDetails.tableName)}
+                    and nspname = ${this.escape(tableDetails.schema)}
+              ) AS schema ON(schema.oid = main.conrelid)
+
+      WHERE
+              main.contype = 'f'
+
+      GROUP BY
+              domain,
+              "tableSchema",
+              "tableName",
+              "constraintType",
+              "constraintSchema",
+              "constraintName",
+              "constraintDefinition",
+              on_update,
+              on_delete,
+              match,
+              "referencedTableSchema",
+              "referencedTableName",
+              "referencedTableConstraintName"
+      ${columnName ? `
+      HAVING
+              "tableColumnNames" && {${this.escape(columnName)}}::text[]
+      ` : ''}
+
+      ORDER BY
+              "domain",
+              "tableSchema",
+              "tableName",
+              "constraintName"
+      ;
+    `.replace(/--.*/g, ''); // remove comments as they would interfere in a single-line statement
```

Comment:
removing comments can be done in `minifySql` instead (`minifySql` being the replacement for `toSingleLine` I proposed in another comment)

Reply:
@ephys are you proposing to move `minifySql` to Utils class instead of Support? It's currently only available in the _test_ environment.

Reply:
https://github.com/sequelize/sequelize/pull/14578/files#r889674953

Yes. You already recommended that :) D'oh.  I need to get more sleep.